### DISCUSSION TITLE
Practical usefulness of removed "can_send_message" after release 0.0.12

### DISCUSSION THREAD

**Discussion by juzpermsky** [lucaspoffo/renet#98] [Q&A] [answered]
You decided to remove that functionality. But it seems to have at least one useful purpose.

Let's consider the case of sending a large file (about 1-2 GB) in ReliableOrdered channel.

The sender could open the file for reading, and iteratively read the contents of the file in small chunks, say 256 KB.
In this case, it would be useful to call can_send_message (to check that ```channel.max_memory_usage_bytes - channel.memory_usage_bytes >= chunk_size```) before calling send_message.
If there is free space in the channel, we can safely move forward the read cursor of the file to get new chunk followed by a call to send_message without fear of disconnection (with reason ```ChannelError::ReliableChannelMaxMemoryReached``` ).

Right now the only way to avoid this error in cases like this, is to increase ```channel_config.max_memory_usage_bytes``` to the size of the file being sent.
But this seems like an inefficient solution, because almost the entire file at the very beginning will be read till the end to the RAM, instead of gradually reading it from the file system in small chunks.

**Comment by lucaspoffo** [comment, marked as answer]
Yeah, it makes sense.
I can see the usefulness. For example, if you have a channel only for big files (ex: content download), then checking if it is possible to send the next file would be ideal instead of sending all of them. Or do some kind of streaming as you described.

I can add back it back, and the check is simple like you said: `channel.max_memory_usage_bytes <= message_size +  channel.memory_usage_bytes`.

**Reply by juzpermsky** [reply]
That would be great. Thank you.

**Reply by lucaspoffo** [reply]
@juzpermsky added `can_send_message` and `channel_available_memory` in edec20e4a2a9bc003375de5a7ffad4f4081c198b
